### DISCUSSION TITLE
Regarding the inability to use statement as an input type in blockly_json within Blockly.

### DISCUSSION THREAD

**Discussion by Lafatalite** [pyri-project/pyri-core#17] [Q&A]
I want to create a block of the `statement` type, but it's currently not working. It gives the error: `unsupported block argument type input_statement used in block global_variable_delete. Use Python generator function instead of sandbox_function_name or sandbox_function_name_selector.` I want to create a function block for establishing a socket, and I want to create a statement block so that after the operation is complete, it can automatically close the connection. However, the `statement` is not working right now.

**Comment by johnwason** [comment]
The pyri sandbox runs "restricted Python" which removes most of the functionality of Python in favor of simplicity. This is intended to make it easier for novice programmers and to make the code more verifiable. We have plans for "user modules" that would be a full Python implementation, but that hasn't been developed yet. For now use sandbox function plugins for functionality that is beyond what is available in the restricted Python.

https://github.com/pyri-project/pyri-core/blob/master/doc/plugin_development.md#sandbox-plugins

https://github.com/pyri-project/pyri-core/blob/master/doc/pyri_dialect_specification.md
